core/build_playerglobal/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+//! An internal Ruffle utility to build our playerglobal
+//! `library.swf`
+
+use std::collections::HashSet;
+use std::fs::File;
+use std::path::PathBuf;
+use std::process::Command;
+use swf::DoAbc;
+use swf::Header;
+use swf::SwfStr;
+use swf::Tag;
+use walkdir::WalkDir;
+
+/// If successful, returns a list of paths that were used. If this is run
+/// from a build script, these paths should be printed with
+/// cargo:rerun-if-changed
+pub fn build_playerglobal(
+    repo_root: PathBuf,
+    out_dir: PathBuf,
+) -> Result<Vec<PathBuf>, Box<dyn std::error::Error>> {
+    let mut classes_dir = repo_root.clone();
+    classes_dir.push("core/src/avm2/globals/");
+
+    let mut asc_path = repo_root.clone();
+    asc_path.push("core/build_playerglobal/asc.jar");
+
+    drop(repo_root); // Appease clippy
+
+    let mut out_path = out_dir;
+    out_path.push("playerglobal.swf");
+
+    let mut classes = Vec::new();
+    for entry in WalkDir::new(&classes_dir) {
+        let entry = entry?;
+        if entry.path().extension().and_then(|e| e.to_str()) == Some("as") {
+            classes.push(entry.into_path());
+        }
+    }
+
+    // These classes are currently stubs - they're referenced by
+    // other classes that we need to compile, but the real definition
+    // is in Ruffle itself (in Rust code).
+    // As a result, we skip including them in our built SWF.
+    let skip_classes: HashSet<_> = ["Object", "Number", "Boolean", "String"]
+        .into_iter()
+        .collect();
+
+    let mut class_bytes = Vec::new();
+    for (i, class) in classes.iter().enumerate() {
```

Comment:
Can you please remind me why we need to compile an `.abc` file per class rather than having a single `.abc` that contains them all?

Reply:
I thought that this was necessary in order for `asc.jar` to work correctly, but it turns out to work fine with just a single invocation. I've updated the PR to do this.